app/javascript/mastodon/features/compose/components/compose_form.jsx

```diff
@@ -41,8 +43,14 @@ const messages = defineMessages({
   saveChanges: { id: 'compose_form.save_changes', defaultMessage: 'Save changes' },
 });
 
+const mapStateToProps = state => ({
+  server: state.getIn(['server', 'server']),
```

Comment:
Since it's only ever used for `max_characters`, the following would probably make more sense:
```suggestion
  maxCharacters: state.getIn(['server', 'server', 'configuration', 'statuses', 'max_characters'], 500),
```